test/gtest/ucm/cuda_hooks.cc

```diff
@@ -63,49 +25,50 @@ class cuda_hooks : public ucs::test {
             UCS_TEST_SKIP_R("can't init cuda device");
         }
 
-        ret = cuDeviceGet(&device, 0);
+        ret = cuDeviceGet(&m_device, 0);
         if (ret != CUDA_SUCCESS) {
             UCS_TEST_SKIP_R("can't get cuda device");
         }
 
-        ret = cuCtxCreate(&context, 0, device);
+        ret = cuCtxCreate(&m_context, 0, m_device);
         if (ret != CUDA_SUCCESS) {
             UCS_TEST_SKIP_R("can't create cuda context");
         }
 
-        memset(&alloc_event, 0, sizeof(alloc_event));
-        memset(&free_event, 0, sizeof(free_event));
+        /* Avoid memory alocation in event callbacks */
+        m_alloc_events.reserve(100);
```

Comment:
we should `.clear()` here to make sure each test case starts with empty events?

Reply:
each test is running in an new instance of the test class so no need